cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -121,6 +121,21 @@ def _validate(self, region):
             )
 
 
+class SchedulerDisableSudoAccessForDefaultUserValidator(Validator):
+    """
+    Validator for DisableSudoAccessForDefaultUser and AWS Batch Scheduler.
+
+    Fail if using AWS Batch and DisableSudoAccessForDefaultUser is not supported.
+    """
+
+    def _validate(self, scheduler):
+        if scheduler == "awsbatch":
+            self._add_failure(
+                "DisableSudoAccessForDefaultUser is not supported when using AWS Batch as scheduler.",
```

Comment:
I think we can solve the potential confusion with public docs saying that in case of AWS Batch scheduler the removal of sudo access is applied only on the head node

Reply:
I re-checked the [code](https://github.com/aws/aws-parallelcluster/blob/2481f621bef8d4967a92b3b5dab53b12e63e3c1c/cli/src/pcluster/templates/awsbatch_builder.py#L159-L160) and I see that we can add the user-data.sh for Ec2 user in AWSBatch](https://docs.aws.amazon.com/batch/latest/userguide/launch-templates.html). However, I can add this in the next PR as I would need to test the change for AWSBatch.